plugins/security/nexus-ldap-plugin/src/main/java/org/sonatype/security/realms/ldap/internal/restlet/AbstractLdapPlexusResource.java

```diff
@@ -177,53 +202,70 @@ protected LdapServerConfigurationDTO toDto(CLdapServerConfiguration ldapServer)
-    return this.convert(userAndGroupConfig, new LdapUserAndGroupAuthConfigurationDTO());
+  protected LdapUserAndGroupAuthConfigurationDTO toDto(Mapping mapping) {
+    final LdapUserAndGroupAuthConfigurationDTO result = new LdapUserAndGroupAuthConfigurationDTO();
+    result.setEmailAddressAttribute(mapping.getEmailAddressAttribute());
+    result.setLdapGroupsAsRoles(mapping.isLdapGroupsAsRoles());
+    result.setGroupBaseDn(mapping.getGroupBaseDn());
+    result.setGroupIdAttribute(mapping.getGroupIdAttribute());
+    result.setGroupMemberAttribute(mapping.getGroupMemberAttribute());
+    result.setGroupMemberFormat(mapping.getGroupMemberFormat());
+    result.setGroupObjectClass(mapping.getGroupObjectClass());
+    result.setUserPasswordAttribute(mapping.getUserPasswordAttribute());
+    result.setUserIdAttribute(mapping.getUserIdAttribute());
+    result.setUserObjectClass(mapping.getUserObjectClass());
+    result.setLdapFilter(mapping.getLdapFilter());
+    result.setUserBaseDn(mapping.getUserBaseDn());
+    result.setUserRealNameAttribute(mapping.getUserRealNameAttribute());
+    result.setUserSubtree(mapping.isUserSubtree());
+    result.setGroupSubtree(mapping.isGroupSubtree());
+    result.setUserMemberOfAttribute(mapping.getUserMemberOfAttribute());
+    return result;
   }
 
-  protected CConnectionInfo toLdapModel(LdapConnectionInfoDTO connectionDto)
-      throws ResourceException, InvalidConfigurationException
+  protected Connection toLdapModel(LdapConnectionInfoDTO from)
+      throws ResourceException
   {
-    return decode(convert(connectionDto, new CConnectionInfo()));
-  }
-
-  private <T, F> T convert(F from, T to) {
-    // xstream cheat ahead
-    XStream xstream = new XStream();
-    xstream.setClassLoader(this.getClass().getClassLoader());
-    String fromXml = xstream.toXML(from);
-    to = (T) xstream.fromXML(fromXml, to);
-    return to;
+    try {
+      final Connection to = new Connection();
+      to.setSearchBase(from.getSearchBase());
+      to.setSystemUsername(from.getSystemUsername());
+      to.setSystemPassword(from.getSystemPassword());
+      to.setAuthScheme(from.getAuthScheme());
+      final Host host = new Host(
+          Protocol.valueOf(from.getProtocol()),
+          from.getHost(),
+          from.getPort()
+      );
+      to.setHost(host);
+      if (!Strings.isNullOrEmpty(from.getBackupMirrorHost())) {
+        final Host backupHost = new Host(
+            Protocol.valueOf(from.getBackupMirrorProtocol()),
+            from.getBackupMirrorHost(),
+            from.getBackupMirrorPort()
+        );
+        to.setBackupHost(backupHost);
+      }
+      to.setSaslRealm(from.getRealm());
+      to.setConnectionTimeout(from.getConnectionTimeout());
+      to.setConnectionRetryDelay(from.getConnectionRetryDelay());
+      to.setCacheTimeout(from.getCacheTimeout());
+      return decode(to);
+    }
+    catch (NullPointerException e) {
```

Comment:
It should be actually ResourceException (restlet). As IAEx is handled by Restlet as HTTP 400, but NPE is handled as HTTP 500, which is in this case undesirable. Ack, and will fix.